### ISSUE TITLE
Setting date fields using a touch device

### ISSUE THREAD

**Issue by bossloper** [modxcms/evolution#1027]
Using a touch device e.g. iPad it is difficult to set a date field that uses the calendar popup.
e.g. setting publish / unpublish date on a resource.
The calendar popup works but there is no way to close the calendar popup after a date is selected. Touching outside the calendar popup does not close it (which might be the intuitive thought). 

You can navigate to a different tab and it will retain the date selected but either clicking outside the popup or an 'ok' (close) button is needed to close the popup for touch devices.

**Comment by Deesen** [comment]
You mentioned in #1054 it is related to this issue, Did the datepicker update solve it also?

**Comment by bossloper** [comment]
Yes and no. 
+ Enter key now working on Android and iOS. 
-  Clicking outside the pop calendar closes the calendar for Andriod devices but not iPad/iPhone

**Comment by Deesen** [comment]
Hmm I have no Apple-device to test it myself. Could you try replacing this line https://github.com/modxcms/evolution/blob/develop/manager/media/style/common/moo-datepicker/datepicker.js#L143 with

    var clickEvent = document.ontouchstart !== null ? 'touchstart' : 'mousedown';
    document.addEvent(clickEvent, this.close.bind(this));

**Comment by bossloper** [comment]
Tried that, but no, not working. + from what I read replacing mousedown with touchstart is not a good idea if an iPad has a mouse attached. Don't think this has ever worked fully on iOS. Maybe one to leave for now for if/when the calendar popup gets updated to a jQuery version?

**Comment by Deesen** [comment]
Thanks for testing! Hmm, but there must be a simple JS-solution to catch this event and close datepicker. It is a pity I cannot test it here myself.

**Comment by Deesen** [comment]
Ok I investigated a little bit further and found [this link](https://www.html5rocks.com/en/mobile/touchandmouse/). So it should be solved easily by adding both events, which will cancel each other out to avoid double triggering. I propose additional event-listener in line 144

    document.addEvent('mousedown', this.close.bind(this));
    document.addEvent('touchstart', this.close.bind(this)); // Added

function close() line 117

    e = new Event(e);
    e.stop();         // Added to avoid double trigger

**Comment by bossloper** [comment]
I tried using the above changes to the original file. It does not help and the 'stop' prevents the time being set on any browser including PC.

**Comment by Deesen** [comment]
With a device at hand to check myself I would find a solution within minutes I´m sure. But like this is frustrating, so maybe I better leave that to someone else. Last thing you could try maybe would be put the stop() before closing } like

    close: function(e) {
      // everything else

      e.stop();
    },

**Comment by Deesen** [comment]
Or like you proposed a simple OK-button beside the time-input could be a saver.

**Comment by bossloper** [comment]
I tried your final code change, just the same. 
I also tried putting an alert() into the close function and it is just not getting triggered in iOS when clicking outside the popup so it is something to do with touchstart and the binding.

But yes I'd leave it for now, there are more important issues. At least by selecting the time and clicking 'GO' in the device keyboard it does close the popup now.
